plugins/kotlin/j2k/shared/src/org/jetbrains/kotlin/nj2k/JavaToJKTreeBuilder.kt

```diff
@@ -904,6 +905,24 @@ class JavaToJKTreeBuilder(
                 it.withFormattingFrom(this)
             }
 
+        private fun PsiMethod.isOverrideMethodWithRedundantVisibility(): Boolean {
+            val superMethods = findSuperMethods()
+            if (superMethods.isEmpty()) return false // Unknown super method
+            val methodVisibility = visibility().visibility
+
+            // ensure none of the found super methods violate redundancy conditions
+            return superMethods.all { superMethod ->
+                val superVisibility = superMethod.visibility().visibility
+                when {
+                    name == "clone" && superMethod.containingClass?.name == "Object"
+                            && superMethods.size == 1 -> methodVisibility == PROTECTED // special built-in type
+                    superVisibility == methodVisibility -> true
+                    superVisibility == INTERNAL -> methodVisibility == PUBLIC
```

Comment:
I noticed that in `K2JavaToKotlinConverterSingleFileTestGenerated.ToKotlinClasses#testMutableIterator`, the overridden methods of the public `java.util.Iterator` are converted to `internal`, but actually they should be public. It looks like this is a bug in `org.jetbrains.kotlin.nj2k.PsiUtilsKt#visibility` for members with default visibility. Can you please investigate how we can do better here? Probably there is some method to compute the default visibility of Java declarations somewhere in the Java plugin that we can reuse.

Also, what happens if our method overrides a Kotlin method?